pactflow/client/ai.ts

```diff
@@ -131,14 +131,42 @@ export const EndpointMatcherSchema = z
     "REQUIRED: Matcher to specify which endpoints from the OpenAPI document to generate tests for. At least one matcher field must be provided"
   );
 
+export const RemoteOpenAPIDocumentSchema = z
+  .object({
+    authToken: z
+      .string()
+      .describe(
+        "Auth Bearer Token if the OpenAPI spec requires authentication."
+      )
+      .optional(),
+    authScheme: z
+      .string()
+      .describe(
+        "Authentication scheme (e.g., 'Bearer', 'Basic'). Default scheme passed should be Bearer if authToken is specified and this field is not set."
+      )
+      .default("Bearer")
+      .optional(),
+    url: z
+      .string()
+      .url("Must be a valid openapi url")
+      .describe("URL of the remote OpenAPI document.")
+      .optional(),
+  })
+  .describe("Use this schema to fetch openapi documents present over a url.");
+
 export const OpenAPIWithMatcherSchema = z
   .object({
-    document: OpenAPISchema,
+    document: OpenAPISchema.describe(
+      "The OpenAPI document describing the API being tested. if document is not provided, don't add the field if remoteOpenAPIDocument is provided."
+    ).optional(),
     matcher: EndpointMatcherSchema,
+    remoteDocument: RemoteOpenAPIDocumentSchema.optional().describe(
+      "The remote OpenAPI document to use for the review/generation in case openapi document is not provided. If provided do not include the document field under openapi."
+    ),
   })
   .describe(
     "If provided, the OpenAPI document which describes the API being tested and is accompanied by a matcher which will be used to identify the interactions in the OpenAPI document which are relevant to the Pact refinement process."
-  );
+  ).transform(addOpenAPISpecToSchema);
```

Comment:
Nice! I think that's a much better solution as it ensures we have a valid structure all the time 🚀 I can see the benefit in where it is used, the code is a lot cleaner.